packages/backend/src/open_payments/wallet_address/routes.ts

```diff
@@ -1,48 +1,60 @@
 import { AccessAction } from '@interledger/open-payments'
 import { WalletAddressSubresource } from './model'
-import { WalletAddressSubresourceService } from './service'
-import { WalletAddressContext, ListContext } from '../../app'
+import {
+  WalletAddressService,
+  WalletAddressSubresourceService
+} from './service'
+import { WalletAddressUrlContext, ListContext } from '../../app'
 import {
   getPageInfo,
   parsePaginationQueryParameters
 } from '../../shared/pagination'
 import { OpenPaymentsServerRouteError } from '../route-errors'
+import { IAppConfig } from '../../config/app'
 
 interface ServiceDependencies {
-  authServer: string
-  resourceServer: string
+  config: IAppConfig
+  walletAddressService: WalletAddressService
 }
 
 export interface WalletAddressRoutes {
-  get(ctx: WalletAddressContext): Promise<void>
+  get(ctx: WalletAddressUrlContext): Promise<void>
 }
 
 export function createWalletAddressRoutes(
   deps: ServiceDependencies
 ): WalletAddressRoutes {
   return {
-    get: (ctx: WalletAddressContext) => getWalletAddress(deps, ctx)
+    get: (ctx: WalletAddressUrlContext) => getWalletAddress(deps, ctx)
   }
 }
 
-// Spec: https://docs.openpayments.guide/reference/get-public-account
 export async function getWalletAddress(
   deps: ServiceDependencies,
-  ctx: WalletAddressContext
+  ctx: WalletAddressUrlContext
 ): Promise<void> {
-  if (!ctx.walletAddress) {
+  // The instance's wallet address is only relevant for the /jwks.json route, so "hide" it when directly requested
```

Comment:
Does this mean we just shouldn't have a `/GET` route for the wallet address? If we still need this route then when would we need it? Comment seems to suggest we only need the `/jwks.json` route IMO but maybe im not following.

Reply:
When I call the `Examples > Open Payments > Get receiver wallet address` bruno request (which creates the wallet address) then call `Open Payments APIs > Get a Wallet Address` I receive the the wallet address.

If I copy/paste the url into the browser `http://localhost:4000/accounts/asmith` then I see `Could not get wallet address`.

How should it be working? It seems like the bruno request should not return the wallet  address according to this comment.

Reply:
@BlairCurrey

`deps.config.walletAddressUrl` is the configured wallet address for the Rafiki instance itself, i.e. its the wallet address used to get incoming payments/incoming payment grants when creating "remote" incoming payments at other Rafikis. e.g. https://ilp.rafiki.money/.well-known/pay/. 

Maybe I can word this comment better, but we actually don't want to resolve https://ilp.rafiki.money/.well-known/pay/ (this shouldn't be a wallet address which resources can be created on, its not for an actual user account). As a result, we only care about the https://ilp.rafiki.money/.well-known/pay/jwks.json path, since it gets looked up to check the keys of the rafiki instance when validating the grant on another rafiki.

Reply:
All other "normal" wallet addresses, should get resolved as usual with the correct asset, publicName, auth/resourceServer urls (since they represent actual user accounts)

Reply:
Thank you for the context.

Same result with `http://localhost:3000/accounts/gfranklin` in the browser, but not with bruno. This is not the case on main - this request works as expected in the browser. The only difference between bruno and the browser I see is the host header that gets added in the pre-request script. Not sure if that's the actual difference.

Here is the error:

```
{"level":30,"time":1715787978022,"pid":1,"hostname":"cloud-nine-wallet-backend","method":"GET","path":"/accounts/gfranklin","message":"Could not get wallet address","msg":"Received error when handling SPSP request"}
```

Looks like its failing at the `if (!walletAddress?.isActive) {` check of the `spspsMiddleware`.

Reply:
going back just 1 step further it looks like `if (ctx.accepts('application/spsp4+json')) {` resolves to false with the bruno request, and true for the browser.

Reply:
What happens when we call `http://localhost:3000/accounts/gfranklin` in the browser, or without the host header in bruno, is we actually end up updating the wallet address via the `wallet_address.not_found` webhook event. (ie if you request `http://localhost:3000/accounts/gfranklin` on main, the wallet address will end up changing from `https://cloud-nine-wallet-backend/accounts/gfranklin`  to `http://localhost:3000/accounts/gfranklin`. 

This PR is different for SPSP request since we use `walletAddressService.getByUrl` instead of `walletAddressService.getOrPollByUrl`. However, I think this PR has same behaviour when we request a wallet address _without_ going through SPSP as well just by requesting the wallet address with `Accept: application/json`. I think we should update the MASE code to prevent this, and seems like a separate issue in general.

I'm not sure why I decided to use `walletAddressService.getByUrl`  in SPSP instead of polling (maybe I assume if you are using SPSP then we should already have a wallet address)?

This makes me wonder if we should default to returning the Open Payments wallet address actually, and only return the SPSP details if explicitly requested?

Reply:
> This makes me wonder if we should default to returning the Open Payments wallet address actually, and only return the SPSP details if explicitly requested?

Chatted w Sabine regarding this, decision made to return Open Payments wallet address by default when fetching it. Also made a change to return SPSP wallet address only if `'application/spsp4+json'` is present in accept header explicitly.